src/XRP/issued-currency-client.ts

```diff
@@ -379,4 +385,163 @@ export default class IssuedCurrencyClient {
+  // TODO: (acorso) consider using an object for potentially long list of params
+  public async sendIssuedCurrency(
+    sender: Wallet,
+    destinationAddress: string,
+    currency: string,
+    issuer: string,
+    value: string,
+    transferFee?: number,
+  ): Promise<TransactionResult> {
+    if (!XrpUtils.isValidXAddress(destinationAddress)) {
+      throw new XrpError(
+        XrpErrorType.XAddressRequired,
+        'Destination address must be in X-address format.  See https://xrpaddress.info/.',
+      )
+    }
+
+    // TODO: (acorso) we don't need to convert back to a classic address once the ripple-binary-codec supports X-addresses for issued currencies.
+    const issuerClassicAddress = XrpUtils.decodeXAddress(issuer)
+    if (!issuerClassicAddress) {
+      throw new XrpError(
+        XrpErrorType.XAddressRequired,
+        'Issuer address must be in X-address format.  See https://xrpaddress.info/.',
+      )
+    }
+    if (!issuerClassicAddress.address) {
+      throw new XrpError(
+        XrpErrorType.XAddressRequired,
+        'Decoded classic address is missing address field.',
+      )
+    }
+
+    const currencyProto = new Currency()
+    currencyProto.setName(currency)
+
+    const issuerAccountAddress = new AccountAddress()
+    issuerAccountAddress.setAddress(issuerClassicAddress.address)
+
+    const issuedCurrency = new IssuedCurrencyAmount()
+    issuedCurrency.setCurrency(currencyProto)
+    issuedCurrency.setIssuer(issuerAccountAddress)
+    issuedCurrency.setValue(value)
+
+    const currencyAmount = new CurrencyAmount()
+    currencyAmount.setIssuedCurrencyAmount(issuedCurrency)
+
+    const amount = new Amount()
+    amount.setValue(currencyAmount)
+
+    const destinationAccountAddress = new AccountAddress()
+    destinationAccountAddress.setAddress(destinationAddress)
+
+    const destination = new Destination()
+    destination.setValue(destinationAccountAddress)
+
+    // Construct Payment fields
+    const payment = new Payment()
+    payment.setDestination(destination)
+    // Note that the destinationTag doesn't need to be explicitly set here, because the ripple-binary-codec will decode this X-Address and
+    // assign the decoded destinationTag before signing.
+    payment.setAmount(amount)
+
+    if (!(transferFee === undefined)) {
+      // Construct SendMax - value should be intended amount + relevant transfer fee
+      // Note that a transfer fee doesn't apply if the source address (of this transaction) and the issuing address of the currency being
+      // sent are the same (i.e. issuer sending currency directly) OR if the issuer of the currency and the destination are the same
+      // (i.e. redeeming an issued currency).
+      // However, it also doesn't hurt to include a SendMax field, it just won't nd up being used (added this to questions just to make sure).
+
+      const numericValue = Number(value)
+      const sendMaxValue = Math.ceil((1 + transferFee / 100) * numericValue)
```

Comment:
can you leave a comment explaining why you're doing this calculation

Reply:
yar

Reply:
this was good actually, I made this way more precise by revisiting.  The original implementation would have potentially created `sendMax` values that were much larger than necessary.  Even though it actually shouldn't typically matter if the `sendMax` value is much larger than what will actually be sent (because rippled will only apply the relevant transfer fees and take no more than that) - it could be a problem if the `sendMax` value is unreasonably high AND someone is trying to make a cross-currency payment that may involve some offers in the DEX that have unreasonably high transfer fees... it would just gobble up those expensive offers because it CAN because it has a high `sendMax`, potentially unbeknownst to the user.

Reply:
with this implementation, the `sendMax` value should be as tight as possible while still achieving its purpose